compiler/optimizer/OMRTransformUtil.cpp

```diff
@@ -394,3 +394,57 @@ OMR::TransformUtil::transformCallNodeToPassThrough(TR::Optimization* opt, TR::No
    node->removeAllChildren();
    node = TR::Node::recreateWithoutProperties(node, TR::PassThrough, 1, child);
    }
+
+void
+OMR::TransformUtil::createConditionalAlternatePath(TR::Compilation* comp,
+                                                   TR::TreeTop *ifTree,
+                                                   TR::TreeTop *thenTree,
+                                                   TR::Block* elseBlock,
+                                                   TR::Block* mergeBlock,
+                                                   TR::CFG *cfg,
+                                                   bool markCold)
+   {
+   cfg->setStructure(0);
+
+   TR::Block* ifBlock = elseBlock;
+   ifBlock->prepend(ifTree);
+   elseBlock = ifBlock->split(ifTree->getNextTreeTop(), cfg, false /*fixupCommoning*/, true /*copyExceptionSuccessors*/);
+   elseBlock->setIsExtensionOfPreviousBlock(true);
```

Comment:
Not every optimization can cope with extended basic blocks in the compiler. What happens if this routine is called before block extension can be tolerated given that the above line sets "elseBlock" to be an extension of the previous block and does not fix up commoning.

Reply:
I see. So no nodes were actually being kept commoned across the 2 blocks anyway.

Reply:
Yes